src/rvoice/fluid_rvoice_mixer.c

```diff
@@ -642,10 +642,20 @@ void delete_fluid_rvoice_mixer(fluid_rvoice_mixer_t* mixer)
 
 
 #ifdef LADSPA				    
-void fluid_rvoice_mixer_set_ladspa(fluid_rvoice_mixer_t* mixer, 
-				   fluid_LADSPA_FxUnit_t* ladspa)
+void fluid_rvoice_mixer_set_ladspa(fluid_rvoice_mixer_t* mixer, fluid_ladspa_fx_t *ladspa_fx)
 {
-  mixer->LADSPA_FxUnit = ladspa;
+  mixer->ladspa_fx = ladspa_fx;
+}
+
+void fluid_rvoice_mixer_deactivate_ladspa(fluid_rvoice_mixer_t *mixer)
+{
+    if (mixer->ladspa_fx == NULL) return;
+    mixer->ladspa_fx->state = FLUID_LADSPA_INACTIVE;
+
+    /* Signal LADSPA engine that deactivation has been processed */
+    fluid_cond_mutex_lock(mixer->ladspa_fx->state_mutex);
```

Comment:
mutex lock in `rvoice_mixer`. Is that a good idea?

Reply:
I wondered as well, but the old code did exactly that in fluid_LADSPA_run. I don't think it's a problem, because the mutex is only used for the cond signalling. So it's going to be a wait-free lock in almost 100% of cases. And that code is not called very often, currently only if a user enters ladsap_reset or ladspa_stop into the shell.

Reply:
Agreed, ok then.